prov/usnic/configure.m4

```diff
@@ -1,124 +1,319 @@
+		[AC_HELP_STRING([--with-libnl3(=DIR)],
+			[Directory prefix for libnl3 (mutually exclusive with --with-libnl)])],
+		[], [with_libnl3=default])
+
+	# The --with options carry two pieces of information: 1) do
+	# you want a specific version of libnl, and 2) where that
+	# version of libnl lives.  For simplicity, let's separate
+	# those two pieces of information.
+	USNIC_PARSE_WITH([libnl], [$with_libnl])
+	USNIC_PARSE_WITH([libnl3], [$with_libnl3])
+
+	# It is erroneous to request both libnl and libnl3
+	AS_IF([test $usnic_want_libnl -eq 1 &&
+	       test $usnic_want_libnl3 -eq 1],
+	      [AC_MSG_WARN([You cannot specify both --with-libnl and --with-libnl3])
+	       AC_MSG_ERROR([Cannot continue])])
+
+	# Default to a numeric value (this value gets AC_DEFINEd)
+	HAVE_LIBNL3=0
+
+	###################################################
+	# NOTE: We *must* check for libnl3 before libnl.
+	###################################################
+
+	# Do we want to check for libnl3?  Note: we check for libnl3
+	# if libnl3 was explicitly requested, or if neither
+	# libnl/libnl3 was specifically requested.  We do *not* check
+	# for libnl3 is libnl was specifically requested.
+	usnic_check_libnl3=0
+	AS_IF([test $usnic_want_libnl3 -eq 1],
+	      [usnic_check_libnl3=1])
+	AS_IF([test $usnic_want_libnl3 -eq -1],
+	      [AS_IF([test $usnic_want_libnl -eq 0 ||
+		      test $usnic_want_libnl -eq -1],
+		     [usnic_check_libnl3=1])])
+
+	AS_IF([test $usnic_check_libnl3 -eq 1],
+	      [USNIC_CHECK_LIBNL3([$usnic_libnl3_location], [usnic_nl])])
+
+	AS_IF([test $usnic_want_libnl3 -eq 1 &&
+	       test "$usnic_nl_LIBS" = ""],
+	      [AC_MSG_WARN([--with-libnl3 specified, but not found])
+	       AC_MSG_ERROR([Cannot continue])])
+
+	# Do we want to check for libnl?  Note: we check for libnl if
+	# the user requested it, or if we're in a default case and we
+	# didn't find libnl3, above.
+	usnic_check_libnl=0
+	AS_IF([test $usnic_want_libnl -eq 1],
+	      [usnic_check_libnl=1])
+	AS_IF([test $usnic_want_libnl -eq -1 &&
+	       test "$usnic_nl_LIBS" = ""],
+	      [usnic_check_libnl=1])
+
+	AS_IF([test $usnic_check_libnl -eq 1],
+	      [USNIC_CHECK_LIBNL([$usnic_libnl_location], [usnic_nl])])
+
+	AS_IF([test $usnic_want_libnl -eq 1 &&
+	       test "$usnic_nl_LIBS" = ""],
+	      [AC_MSG_WARN([--with-libnl specified, but not found])
+	       AC_MSG_ERROR([Cannot continue])])
+
+	# Final result
+	AC_SUBST(HAVE_LIBNL3)
+	AC_DEFINE_UNQUOTED([HAVE_LIBNL3], [$HAVE_LIBNL3],
+	      [Whether we have libl or libnl3])
+
+	AC_SUBST(usnic_nl_CPPFLAGS)
+	AC_SUBST(usnic_nl_LDFLAGS)
+	AC_SUBST(usnic_nl_LIBS)
+
+	AS_IF([test "$usnic_nl_LIBS" = ""],
+	      [usnic_happy=0])
+])
+
+dnl
+dnl Check for libnl-3.
 dnl
-dnl Check for libnl; prefer version 3 instead of version 1.  Abort (i.e.,
-dnl AC_MSG_ERROR) if neither libnl v1 or v3 can be found.
+dnl Inputs:
+dnl
+dnl $1: prefix where to look for libnl-3
+dnl $2: prefix of _CPPFLAGS and _LDFLAGS and _LIBS
```

Comment:
might want to say "variable name prefix" instead of just "prefix", otherwise seems confusing when compared to the preceding line